src/main.sh

```diff
@@ -34,21 +34,56 @@ function main::exec_tests() {
 
 function main::exec_assert() {
   local original_assert_fn=$1
-  local assert_fn=$original_assert_fn
   local args=("${@:2}")
 
+  local assert_fn=$original_assert_fn
+
+  # Check if the function exists
   if ! type "$assert_fn" > /dev/null 2>&1; then
-    # try again using prefix `assert_`
     assert_fn="assert_$assert_fn"
     if ! type "$assert_fn" > /dev/null 2>&1; then
-      echo "Function $original_assert_fn does not exist."
+      # Write into stderr
+      echo "Function $original_assert_fn does not exist." 1>&2
       exit 127
     fi
   fi
 
-  "$assert_fn" "${args[@]}"
+  # Get the last argument safely by calculating the array length
+  local last_index=$((${#args[@]} - 1))
+  local last_arg="${args[$last_index]}"
+  local inner_exit_code=0
+  local bashunit_exit_code=0
+  local output
+
+  if [[ "$assert_fn" == "assert_exit_code" && "$last_arg" == eval* ]]; then
+    local callable_command="${last_arg#eval }"
+    output=$(eval "$callable_command" 2>&1 || echo "inner_exit_code:$?")
```

Comment:
`eval` is a real bash command, but in this context of last argument I am using it at `bashunit` level so hide the complexity and bring familiarity with the "bash context" without bringing new weird terms. This is the trick:
```bash
# remove the "eval" word from the last argument
local callable_command="${last_arg#eval }"

# use real bash "eval" to evaluate your command
output=$(eval "$callable_command" 2>&1 || echo "inner_exit_code:$?")
```